table/tables/partition.go

```diff
@@ -714,6 +748,40 @@ func (lp *ForListPruning) locateListColumnsPartitionByRow(ctx sessionctx.Context
 	return location[0].PartIdx, nil
 }
 
+func (lp *ForListPruning) buildSortedListPartition(ctx sessionctx.Context, tblInfo *model.TableInfo,
```

Comment:
How about put build `btree` logic into `buildListPartitionValueMap` and rename it. there are a lot of duplicate code here.